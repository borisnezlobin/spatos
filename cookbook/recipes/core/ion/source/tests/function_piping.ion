fn square
    read x
    echo $(( x * x ))
end

echo 5 | square

fn square x
    echo $(( x * x ))
end

square 5 | cat

fn format_with pat
    read input
    echo $join(@split(input) $pat)
end

echo one two three four five | format_with "-"

fn square
    read x
    echo $(( x * x ))
end

fn mult x
    read y
    echo $(( y * x ))
end

echo 5 | square | mult 3
