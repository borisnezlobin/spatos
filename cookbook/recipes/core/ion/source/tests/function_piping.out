25
25
one-two-three-four-five
75
